packages/circuits/circuits/hasher.circom

```diff
@@ -3,13 +3,38 @@ include "./circomlib/circuits/poseidon.circom";
 // attester_id, epoch, and nonce must be range checked
 // outside of this component
 template EpochKeyHasher() {
+  var NONCE_BITS = 8;
+  var ATTESTER_ID_BITS = 160;
+  var EPOCH_BITS = 48;
+
   signal input identity_secret;
   signal input attester_id;
   signal input epoch;
   signal input nonce;
+  signal input chain_id;
 
   signal output out;
-  out <== Poseidon(2)([identity_secret, attester_id + 2**160*epoch + 2**208*nonce]);
+
+  /**
+  * control structure
+  * 160 bits attester id
+  * 48 bits epoch
+  * 8 bits nonce
+  * 36 bit chain id
+  **/
+  var acc_bits = 0;
+  var acc_data = attester_id;
```

Comment:
Perhaps we should add a note that the range of these values should be constrained elsewhere.

Reply:
or we can move the constraints to hasher templates?

Reply:
Add notes here, [74ffc42](https://github.com/Unirep/Unirep/pull/547/commits/74ffc4293cca0ae8aad032b819c5855e0cb13087)
I think constraints could be added before using the template (to save constraints)

(the documentation hasn't updated for the hasher templates,
if the documentation is updated, we can also add notes there)